### PR TITLE
Add support for nginx ubi image

### LINKED ISSUE #25: The pipeline builds a UBI-based nginx image for Conjur
## Is your feature request related to a problem? Please describe.

Build an Nginx [UBI-based](https://developers.redhat.com/products/rhel/ubi) image that can be used when running Conjur in OpenShift and can ultimately be published to the RH container registry.

## Describe the solution you would like

Dockerfile with ubi8 base image that builds Nginx with relevant configuration.

### PR DESCRIPTION
**Pull request by guygiat**
### What does this PR do?
- Adds new nginx-ubi folder with all relevant files to build the container.

### What ticket does this PR close?
Resolves #25

### Checklists
- [x] Verify license file is the right one to use.
- [ ] Verify description and summary labels in Dockerfile.
#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] This PR does not require updating any documentation, or
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs

### CODE REVIEW

File: nginx-ubi/config/dhparams.pem [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-----BEGIN DH PARAMETERS-----
```
Comment (izgeri) [inline nginx-ubi/config/dhparams.pem:1]: this seems weird to hardcode this here - I'd expect this to be generated at deploy time (e.g. using `openssl dhparam -out dhparam.pem 4096`). should this be something the helm chart and/or operator is responsible for?

Comment (guygiat) [reply nginx-ubi/config/dhparams.pem:1]: It was weird to me to, but this is how we do it on our helm chart and in our container-dap repo as well.
What should we do?

Comment (sgnn7) [reply nginx-ubi/config/dhparams.pem:1]: This is a pretty complex topic. See [here](https://conjurhq.slack.com/archives/CGTRB6G8G/p1595526917411100) for more info.

Comment (guygiat) [reply nginx-ubi/config/dhparams.pem:1]: It seems that in our helm chart we are using the same DH:
https://github.com/cyberark/conjur-oss-helm-chart/blob/master/conjur-oss/files/dhparams.pem
So what should be the resolution here?

Comment (sgnn7) [reply nginx-ubi/config/dhparams.pem:1]: You can leave it as-is for now or generate on the fly. Generating a key larger than 1024 bits takes a very long time so just using this is better for now. The main important thing is that this file is generated by us and not a baked in-one from a distro but you can make your own with `openssl dhparam -out dhparams.pem 4096`. 

While testing the command, I also created one here too so feel free to use any of the 3 options:
```
-----BEGIN DH PARAMETERS-----
MIICCAKCAgEAlwtBWJfFjCcCFM+/7dS2oB1SwWWQcX7GMsqv4Gd1bFmzIb36ft43
gGFNURJQhplY0LnRWpTIPPChtGicWzTn267YnZ7dS/9g1HaF6Aqy1JZg1bwKY4fs
WOW11pY/W23p9G62DtvteoWmcVC8CF78NKm1PiDXdVc7z1MWH9GK7bJuzjZ7tlK7
Uwr9vOIuSFh/MPrhyW4EsMTEf0StWbLgb+UmfaAnAsExMbpbRNsmjpT/5r0I9UbK
KU287x3ED8T5Q4VfIGkDZFna1rxlZCVbOsYbRcMHXbsHAWYqLGMYgEmS/icXJpm2
RMzO8ne2Z36HqYmbLWCNblg8F2065QfJ9MvH0Y2muTrf4WXUTuVms5gA0ZtXc9hU
aIu8LOzEpJLa6V1/tU2PAGr9Aee9B/G5cV7kHVBjwbVFQMPA0FhjAKIqVI0L1IBa
rZ113AgTTRPpNH5g8A4exOFh6/mcMqI7dmdcfLtvzwHnJdXYDDE1JItdN6zCTWym
lzreYYBYYgyZpaY2n3LOnozyzYaRTGj3uLCvJS+EFRVIInc+8kzkYd7VWUK2+tve
w6W1L3Vx2LUiPWv1NM9disSoH4V2tU9Y7SD6qOM0kzeaQ3h1gIe0V26Uvq/MD1+t
PJHyKPmzAm8V56R2GOPYCldppi/xI3xfIHVHBnDcTWt2v2grkzvmw4sCAQI=
-----END DH PARAMETERS-----
```

Comment (garymoon) [reply nginx-ubi/config/dhparams.pem:1]: Please don't generate dhparams on the fly anywhere in CI. If you need readily accessible dhparams you can pull them out of ConjurOps from [here](https://github.com/conjurinc/conjurops-policies/blob/master/policy/dhparams.yml) (this is new and will be announced on Slack soon).

Comment (guygiat) [reply nginx-ubi/config/dhparams.pem:1]: @izgeri so its seems that we will keep it as is, could you please review the 2 checklist items and update how to proceed?

File: nginx-ubi/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+cd "$(dirname "$0")"
+
+NGINX_VERSION=1.14
```
Comment (izgeri) [inline nginx-ubi/build.sh:5]: @sgnn7 is this consistent with what we use in the helm chart / quick start?

**Review by izgeri — COMMENTED** [review]
I left one comment for @sgnn7 and I'd like him to do a technical review to make sure this PR looks good

Also, I'm marking this as "doesn't need a changelog entry", but we should definitely add one when we publish this image to RH registry.

File: nginx-ubi/build.sh (hunk 2 above)
Comment (sgnn7) [reply nginx-ubi/build.sh:5]: Helm: 1.15
Quick start: 1.13.6

I'd probably opt we go w/ 1.15 to match the helm chart. With that said, NGINX is incredibly stable with APIs so these differences should not matter much (if at all).

Comment (izgeri) [reply nginx-ubi/build.sh:5]: wouldn't hurt to bump up the quick start too.

**Review by sgnn7 — APPROVED** [review]
@guygiat LGTM as a first round implementation though I'd also like to see the NGINX version bumped to 1.15 as @izgeri suggested.

File: nginx-ubi/build.sh (hunk 2 above)
Comment (guygiat) [reply nginx-ubi/build.sh:5]: It seems that ubi8 comes with Nginx 11.4 app stream.
In order to install 1.16 we will need a redhat subscription license (they don't provide 1.15).
Can I keep it with 11.4 at the moment?

Comment (sgnn7) [reply nginx-ubi/build.sh:5]: It should be fine w/ 1.14 but it should be added as a comment that it needs to be kept as close as possible to appliance's NGINX version.

**Comment by sgnn7** [comment]
@guygiat Please also add a comment for your commit and capitalize the title of the commit message

**Review by sgnn7 — APPROVED** [review]
@guygiat LGTM!

**Pull request merged**